connectors/sources/google_cloud_storage.py

```diff
@@ -199,8 +185,54 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+    def _validate_configurations(self):
+        """Validates whether user input is empty or not for configuration field.
+
+        Raises:
+            Exception: Configured keys can't be empty.
+        """
+        if self.configuration["service_account_credentials"] == "":
+            raise Exception("service_account_credentials can't be empty.")
```

Comment:
Could we have a more specific exception like `DataSourceConfigurationError` here, which we'll use in all `_validate_configuration` implementations?

Reply:
Added `DataSourceConfigurationError` in `source.py`, and raised the same with custom messages for the validations.
[bcc60b1](https://github.com/elastic/connectors-python/pull/493/commits/bcc60b1b9b30a0cedc14da979193d02437690281)